src/app

```diff
@@ -0,0 +1,131 @@
+#!/bin/sh
+
+set -e
+
+temp_dir=$(mktemp -d "app.tmp.XXXXXXXX")
+
+cleanup() {
+    set +e
+    test "$temp_dir" != "" && test "$temp_dir" != "/" && rm -Rf "${temp_dir}"
+    test -f "$persistent_store"/.rw_test && rm -f "$persistent_store"/.rw_test
+}
+
+trap cleanup 1 2 3 6 15
+
+config_file="/etc/mender/mender-app.conf"
+version=""
+orchestrator=""
+platform=""
+manifests_dir=""
+app_module_dir="/usr/share/mender/app-modules/v1"
+app_sub_module=""
+artifact_name=""
+persistent_store="/data/mender-app"
+application_name=""
+
+if test -f "$config_file"; then
+    . "$config_file"
+fi
+
+if test -d "$persistent_store"; then
+    if touch "$persistent_store"/.rw_test; then
+        rm -f "$persistent_store"/.rw_test
+    else
+        echo "ERROR: cant write to persistent_store in $persistent_store"
+        exit 1
+    fi
+fi
+
+delta_cmd="xdelta3 -d -s"
```

Comment:
Can we use long options for `-d` and `-s`?

Reply:
as far as I know there are no long options for `xdelta3` for -s and -d.

Reply:
Okey :see_no_evil: